Yosemite/Yosemite/Stores/Order/OrdersUpsertUseCase.swift

```diff
@@ -137,11 +135,10 @@ struct OrdersUpsertUseCase {
     /// Updates, inserts, or prunes the provided StorageOrderItem's taxes using the provided read-only OrderItem
     ///
     private func handleOrderItemTaxes(_ readOnlyItem: Networking.OrderItem, _ storageItem: Storage.OrderItem, _ storage: StorageType) {
-        let itemID = readOnlyItem.itemID
 
         // Upsert the taxes from the read-only orderItem
         for readOnlyTax in readOnlyItem.taxes {
-            if let existingStorageTax = storage.loadOrderItemTax(itemID: itemID, taxID: readOnlyTax.taxID) {
+            if let existingStorageTax = storageItem.taxes?.first(where: { $0.taxID == readOnlyTax.taxID }) {
```

Comment:
Same as the other comment. 

I see that `loadOrderItemTax` method checks for both `itemID` and `taxID`. Do we have to add the additional check of `itemID` here?

Reply:
Similar as above, `storageItem` here is the order item filtered by `itemID` in `handleOrderItems`. Filtering by `itemID` would be redundant.